tools/publish-release.sh

```diff
@@ -0,0 +1,13 @@
+#!/bin/sh
+
+set -ex
+# Note: make sure these are ordered so that dependencies come before the crates that depend on them
```

Comment:
IIUC we never expect this to run in automation, so do you think it makes sense to, say, show the output of `git status` and check the user really does want to publish something? eg, I'm worried someone will execute, say, `publish-release.sh --help`

Reply:
Good point, I added a check that `git status` is clean and also added a confirmation prompt.